RFC-0012-profile-directed-typing.md

```diff
@@ -0,0 +1,295 @@
+## Overview
+
+Missing/incorrect type annotations is a common reason for failed model scripting. For a long time, the process of mitigating missing/incorrect type annotations has been through trial and error, ie. by repeatedly hitting the wall of various type-checking errors generated from torch.jit.script and fixing them one by one. This is both inefficient and frustrating.
```

Comment:
I would prefer "A common reason for failed model scripting is missing or incorrect type annotations", because then the first part of the sentence is sort of a "build up" to the solution we're proposing